Automattic/delphin
# Search: Add and refactor footer

This pull request addresses https://github.com/Automattic/delphin/issues/651 by adding the main menu in the footer of the `Search` page. This was the only page where it was missing. It also refactors this footer to simplify the layouts code and to use a more robust positioning system in order to stick it to the bottom of the page.

![screenshot](https://cloud.githubusercontent.com/assets/594356/19477190/38c960d8-953d-11e6-8032-05cf9c52f764.png)
#### Testing instructions
1. Run `git checkout add/search-page-footer` and start your server, or open a [live branch](https://delphin.live/?branch=add/search-page-footer)
2. Open the [`Search` page](http://delphin.localhost:1337/)
3. Check that the main menu is now displayed at the bottom of the page
4. Check that this menu displays correctly on any other pages as well

You should also check pages that are usually accessed server-side such as the [`Not Found` page](http://delphin.localhost:1337/404) or the `Verify User With Query` page (the page displayed when clicking the link in the login confirmation email). You may also want to check the display on small viewports.
#### Additional notes

Moving to flexbox to position the footer allows us to define in a single place the height (in viewport metrics) of the content of the page. Previously that was something that was set in some layouts (not in every one of them though), resulting in approximate positioning (the padding around the footer was different on some pages, and would vary when minimizing the height of the viewport). In order to simplify the logic and facilitate this transition, I opted to display a border at the top of this footer in small viewports on every page: previously it would only be displayed on certain pages.
#### Merging instructions

**https://github.com/Automattic/delphin/pull/755 is based on this pull request**. As such, https://github.com/Automattic/delphin/pull/755 should be merged first before merging this pull request.
#### Reviews

@Automattic/sdev-feed
